### DISCUSSION TITLE
Is it possible to synchronize the SFTP directory with the command directory?

### DISCUSSION THREAD

**Discussion by crazyweeds** [Eugeny/tabby#5978] [General]
Is it possible to synchronize the SFTP directory with the command directory?

**Comment by Eugeny** [comment]
Yes: https://github.com/Eugeny/tabby/wiki/Shell-working-directory-reporting

**Reply by MrIDK22** [reply]
@Eugeny any further instructions on where to add this for Windows? I've tried adding a `.bash_profile` with the content from the link under my user directory but it doesn't seem to work.

**Reply by Eugeny** [reply]
You only need to add this to the remote side's bash_profile/zshrc

**Reply by MrIDK22** [reply]
So if I have 20 different profiles / connections, I'd need to add it to all 20 for it to work?

**Reply by Eugeny** [reply]
Yes. An SSH client can't "know" what the remote shell's CWD is without shell explicitly reporting. Some other terminals try to hack around this by trying to guess the CWD based on the shell prompt but that's prone to errors

**Reply by MrIDK22** [reply]
Alrighty, thanks Eugeny :)

**Reply by crazyweeds** [reply]
I think it could be implemented in such a way that the shell terminal writes a temporary variable to the current directory each time it cd's. SFTP listens to this temporary variable and goes to the corresponding directory.

**Reply by Eugeny** [reply]
The SSH client can't see the shell variables either. The way it's implemented is via an invisible terminal control sequence that the shell prints.

**Reply by crazyweeds** [reply]
Thank you for letting me know
